lnd.go

```diff
@@ -227,6 +229,29 @@ func lndMain() error {
 	}
 	idPrivKey.Curve = btcec.S256()
 
+	// Connect to router
```

Comment:
This should be gated behind a configuration option. As it may be possible that a user doesn't wish to have their IP address automatically advertised to the network. The parsing of this configuration would happen within `config.go`, allowing a proper `NodeAnnouncement` to be crafted and broadcast to the network. With this patch as is, the discovered external IP address and port mapping wouldn't be advertised at all.